### PR TITLE
[Text Translation] Change scopes to audiences

### PR DESCRIPTION
**Pull request by MikeyMCZ**
# Description

Change `scopes` to `audiences` to align with sdk naming convention.

### CODE REVIEW

**Review by catalinaperalta — APPROVED** [review]

File: sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py [hunk 1]
```diff
@@ -49,8 +49,8 @@ class TranslatorEntraIdAuthenticationPolicy(BearerTokenCredentialPolicy):
     :type credential: ~azure.core.credentials.TokenCredential
```
Comment (catalinaperalta) [inline sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py:49]: Actually we need to update the docs on the policy as well to include the other properties.

Comment (catalinaperalta) [reply sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py:49]: So this means resource_id, region, audience

File: sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py [hunk 2]
```diff
@@ -86,19 +94,21 @@ def set_authentication_policy(credential, kwargs):
     elif hasattr(credential, "get_token"):
         if not kwargs.get("authentication_policy"):
             if kwargs.get("region") and kwargs.get("resource_id"):
+                scope = kwargs.pop("audience", DEFAULT_ENTRA_ID_SCOPE) + DEFAULT_SCOPE
```
Comment (annatisch) [inline sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py:97]: Just to be safe:
```suggestion
                scope = kwargs.pop("audience", DEFAULT_ENTRA_ID_SCOPE).rstrip("/") + DEFAULT_SCOPE
```

File: sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py (hunk 1 above)
Comment (MikeyMCZ) [reply sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py:50]: OK, added.

File: sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py [hunk 3]
```diff
@@ -86,19 +94,21 @@ def set_authentication_policy(credential, kwargs):
...
             else:
                 if kwargs.get("resource_id") or kwargs.get("region"):
                     raise ValueError(
-                        "Both 'resource_id' and 'region' must be provided with a TokenCredential for authentication."
+                        """Both 'resource_id' and 'region' must be provided with a TokenCredential for
+                         regional resource authentication."""
                     )
                 kwargs["authentication_policy"] = BearerTokenCredentialPolicy(
-                    credential, *kwargs.pop("scopes", [DEFAULT_TOKEN_SCOPE]), kwargs
+                    credential, *kwargs.pop("audience", [DEFAULT_TOKEN_SCOPE]), kwargs
```
Comment (annatisch) [inline sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py:111]: Audience is a string - so cannot be used with * syntax here.
I think you would need something like:
`*[kwargs.pop("audience", DEFAULT_TOKEN_SCOPE)]`

Comment (annatisch) [reply sdk/translation/azure-ai-translation-text/azure/ai/translation/text/_patch.py:111]: Also - shouldn't that be `**kwargs` at the end?

**Pull request merged**